### ISSUE TITLE
Missing closing parenthesis deletes code

### ISSUE THREAD

**Issue by i-am-the-slime** [natefaubion/purescript-tidy#91]
I can't quite tell what the real issue is but I just had tidy delete a good chunk of code. The version is `0.7.1`. 
Funnily enough it used `unsafeCoerce` and tidy seemed to take this quite literally:

Here is a small Repro:

Input:
```
module Example where

delete me =
  unsafeCoerce (f
    (a lotta code is actually here
    )
```

Output:
```
module Example where

delete me =
  unsafeCoerce
```

**Comment by JordanMartinez** [comment]
Which version of `purs-tidy` are you on?

**Comment by JordanMartinez** [comment]
Oh, sorry, I missed that part.

**Comment by natefaubion** [comment]
@mikesol experienced this as well, but didn’t have a repro I could test. When I tested the simple missing paren case, it correctly failed with an error. Are you able to repro?

**Comment by natefaubion** [comment]
Now that I'm able to test, the repro does indeed work. Thanks! I suspect something is up in the parser.

**Comment by natefaubion** [comment]
This is fixed in 0.7.2

**Comment by natefaubion** [comment]
The exact issue was an error in the implementation of `many` which propagated the wrong parser state in the case of a failure. This particular case was showing up in the call to `many` for expression applications. Parens and applications are a common case, but it could show up with any parse failure in any call to `many`.
